casepro/backend/junebug.py

```diff
@@ -31,8 +31,20 @@ def get_paginated_response(self, url, params={}, **kwargs):
             # params are included in the next url
             params = {}
 
+    def get_identity(self, uuid):
+        '''Returns the details of the identity.'''
+        r = self.session.get(
+            '%s/api/v1/identities/%s/' % (self.base_url, uuid))
+        if r.status_code == 404:
+            return None
+        return r.json()
+
     def get_addresses(self, uuid):
-        '''Get the list of addresses for an identity specified by uuid.'''
+        '''Get the list of addresses that a message to an identity specified
+        by uuid should be sent to.'''
+        identity = self.get_identity(uuid)
+        if identity and identity.get('communicate_through') is not None:
+            return self.get_addresses(identity['communicate_through'])
```

Comment:
@imsickofmaps said that `communicate_through` should only be followed one step. So we probably need a flag here to say whether we've already followed a `communicate_through` link.

@imsickofmaps Am I correct?